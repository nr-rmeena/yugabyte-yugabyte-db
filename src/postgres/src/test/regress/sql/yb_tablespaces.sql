-- Create Tablespace API tests
-- Create Tablespace should not work if neither LOCATION nor options
-- are specified
CREATE TABLESPACE x;
CREATE TABLESPACE x OWNER yugabyte;

-- Ill formed JSON
CREATE TABLESPACE x WITH (replica_placement='[{"cloud"}]');

-- One of the required keys missing
CREATE TABLESPACE x WITH (replica_placement='[{"cloud":"cloud1","region":"r1","zone":"z1"}]');

-- Invalid value for replication factor
CREATE TABLESPACE x WITH (replica_placement='[{"cloud":"cloud1","region":"r1","zone":"z1","min_number_of_replicas":"three"}]');

-- Positive cases
CREATE TABLESPACE x LOCATION '/data';
CREATE TABLESPACE y WITH (replica_placement='[{"cloud":"cloud1","region":"r1","zone":"z1","min_number_of_replicas":3},{"cloud":"cloud2","region":"r2", "zone":"z2", "min_number_of_replicas":3}]');

-- describe command
\db+
