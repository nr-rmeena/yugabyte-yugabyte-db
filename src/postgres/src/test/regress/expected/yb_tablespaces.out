-- Create Tablespace API tests
-- Create Tablespace should not work if neither LOCATION nor options
-- are specified
CREATE TABLESPACE x;
ERROR:  While creating tablespaces, one of LOCATION or WITH options should be present at or near ";"
LINE 1: CREATE TABLESPACE x;
                           ^
CREATE TABLESPACE x OWNER yugabyte;
ERROR:  While creating tablespaces, one of LOCATION or WITH options should be present at or near ";"
LINE 1: CREATE TABLESPACE x OWNER yugabyte;
                                          ^
-- Ill formed JSON
CREATE TABLESPACE x WITH (replica_placement='[{"cloud"}]');
ERROR:  invalid input syntax for type json
DETAIL:  Expected ":", but found "}".
CONTEXT:  JSON data, line 1: [{"cloud"}...
-- One of the required keys missing
CREATE TABLESPACE x WITH (replica_placement='[{"cloud":"cloud1","region":"r1","zone":"z1"}]');
ERROR:  Key min_number_of_replicas must be defined in the placement policy
-- Invalid value for replication factor
CREATE TABLESPACE x WITH (replica_placement='[{"cloud":"cloud1","region":"r1","zone":"z1","min_number_of_replicas":"three"}]');
ERROR:  Invalid value for "min_number_of_replicas" key
DETAIL:  Found "three" but min_replicas_for_block should be  an integer > 0
-- Positive cases
CREATE TABLESPACE x LOCATION '/data';
WARNING:  LOCATION not supported yet and will be ignored
LINE 1: CREATE TABLESPACE x LOCATION '/data';
                            ^
HINT:  See https://github.com/YugaByte/yugabyte-db/issues/6569. Click '+' on the description to raise its priority
CREATE TABLESPACE y WITH (replica_placement='[{"cloud":"cloud1","region":"r1","zone":"z1","min_number_of_replicas":3},{"cloud":"cloud2","region":"r2", "zone":"z2", "min_number_of_replicas":3}]');
-- describe command
\db+
                                                                                                                                  List of tablespaces
    Name    |  Owner   | Location | Access privileges |                                                                                                Options                                                                                                |   Size    | Description
------------+----------+----------+-------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+-------------
 pg_default | postgres |          |                   |                                                                                                                                                                                                       | 608 bytes |
 pg_global  | postgres |          |                   |                                                                                                                                                                                                       | 43 kB     |
 x          | yugabyte |          |                   |                                                                                                                                                                                                       |           |
 y          | yugabyte |          |                   | {"replica_placement=[{\"cloud\":\"cloud1\",\"region\":\"r1\",\"zone\":\"z1\",\"min_number_of_replicas\":3},{\"cloud\":\"cloud2\",\"region\":\"r2\", \"zone\":\"z2\", \"min_number_of_replicas\":3}]"} |           |
(4 rows)
